pi_controller/README.md

```diff
@@ -0,0 +1,31 @@
+# Raspberry Pi
+
+## Functions
+
+The Raspberry Pi in Cat Town performs a few key functions:
+
+* Runs computer vision to detect the cat's location;
+* Simulates mouse personalities to decide where the mice will run next;
+* Decides when to deploy mice and which ones;
+* Serves a user interface over the Web for remote viewing and control.
+
+## Module Architecture
+
+Cat Town is a monolithic Python program with modules for each basic function. Modules are expected to handle their own threading/subprocesses.
+
+* The **Platform Manager** (`/platform_mgr`) tracks which mice are active, deploys new mice, and handles mouse changes;
+* The Platform Manager runs a set of **Personalities** (`/personalities/*`);
+* The **Computer Vision System** (`/cv`) uses the camera to report the cat's location to Personalities;
+* The **User Interface** is not yet started! See https://feline.atlassian.net/wiki/spaces/FES/pages/34045954/Local+Web+Server for the bounty :D
+
+> todo: where does detecting mouse catch events go in here? CV? platform? maybe a new module/class that uses both of those two? idk
```

Comment:
was personally imaging a few modules:
1) Camera vision:
  - detects cat 
  - detects mouse
  -  reads arco marker -> position + marker code value (that represents which obstacle is installed)  
  -  confirms camera is set up properly (proper distance, angle of view, lighting, obfuscation, etc)
  - returns error codes for camera status  

2) Movement driver:
  -  moves to given coordinates
  - returns error codes for mechanical or movement issues 
  - returns how many slabs (2'x4') are installed/joined
  - (ideally/eventually) can tell us which obstacles are installed and where 
  
3) Joystick Module
  - reads movement from joystick 
  - sends gcode  

4) Main controller 
  - sets state to automatic or manual mode 
  - interfaces with joystick module, camera module, and movement driver.
  - reads gcode from joystick and forwards to movement driver 
  - maintains state of each module n stuff 
  - uses camera module to react to cat 
  - uses board controller to move mouse
  - uses camera module to check if mouse is still attached 
  - given information from camera module and movement driver, return state of operation 
  - error code related to connections (like if we rely on wifi or anything else)
  - propagate errors from either module

Here's a diagram of how I was envisioning this connection 
<img width="668" alt="Screenshot 2024-05-09 at 3 30 16 PM" src="https://github.com/jadebuilds/cattown/assets/19254083/c9c2de54-5dfb-441c-b1d9-e77a03f276d5">

Reply:
but open to changing/iterating this.